pkgs/ok_http/lib/src/ok_http_client.dart

```diff
@@ -0,0 +1,148 @@
+///   }
+/// }
+/// ```
+///
+class OkHttpClient extends BaseClient {
+  late bindings.OkHttpClient _client;
+
+  OkHttpClient() {
+    _client = bindings.OkHttpClient.new1();
+  }
+
+  @override
+  Future<StreamedResponse> send(BaseRequest request) async {
+    var requestUrl = request.url.toString();
+    var requestHeaders = request.headers;
+    var requestMethod = request.method;
+    var requestBody = await request.finalize().toBytes();
+
+    // A Completer to return the response, as OkHttp's enqueue API is async
+    final responseCompleter = Completer<StreamedResponse>();
+
+    // Create an OkHttp Request Builder
+    var reqBuilder = bindings.Request_Builder().url1(requestUrl.toJString());
+
+    requestHeaders.forEach((headerName, headerValue) {
+      reqBuilder.addHeader(headerName.toJString(), headerValue.toJString());
+    });
+
+    // OkHttp doesn't allow a non-null RequestBody for GET and HEAD requests
+    // So, we need to handle this case separately
+    bindings.RequestBody okReqBody;
+    if (requestMethod != 'GET' && requestMethod != 'HEAD') {
+      okReqBody = bindings.RequestBody.create10(requestBody.toJArray());
+    } else {
+      okReqBody = bindings.RequestBody.fromReference(jNullReference);
+    }
+
+    reqBuilder.method(
+      requestMethod.toJString(),
+      okReqBody,
+    );
+
+    // `enqueue()` schedules the request to be executed in the future.
+    // https://square.github.io/okhttp/5.x/okhttp/okhttp3/-call/enqueue.html
+    _client
+        .newCall(reqBuilder.build())
+        .enqueue(bindings.Callback.implement(bindings.$CallbackImpl(
+          onResponse: (bindings.Call call, bindings.Response response) {
+            var responseHeaders = <String, String>{};
+
+            response.headers().toMultimap().forEach((key, value) {
+              responseHeaders[key.toDartString(releaseOriginal: true)] =
+                  value.join(',');
+            });
+
+            int? contentLength;
+            if (responseHeaders.containsKey('content-length')) {
+              contentLength = int.tryParse(responseHeaders['content-length']!);
+
+              // To be conformant with RFC 2616 14.13, we need to check if the
+              // content-length is a non-negative integer
+              if (contentLength == null || contentLength < 0) {
+                responseCompleter.completeError(ClientException(
+                    'Invalid content-length header', request.url));
+                return;
+              }
+            }
+
+            try {
+              responseCompleter.complete(StreamedResponse(
+                Stream.value(response.body().bytes().toUint8List()),
+                response.code(),
+                reasonPhrase:
+                    response.message().toDartString(releaseOriginal: true),
+                headers: responseHeaders,
+                request: request,
+                contentLength: contentLength,
+              ));
+            } catch (e) {
```

Comment:
`response.body().bytes()` seems a general problem without exception handling. Exceptions in code called through JNI need to be caught.
For e.g., 
1. TC: `server headers content length bigger than actual body`
   `java.net.ProtocolException: unexpected end of stream`
   When left un-caught, blocks the execution and the TC stops with `did not complete [E]`. More dangerously, it blocks all other tests from executing as the application crashes immediately because of it :(
   
2. The general use-case of requesting messages which might need long timeouts
   `java.net.SocketTimeoutException: timeout`
   Crashes app with no recovery, when un-caught.
   
Adding a comment for future reference. ✅